lib/reporters/cli/index.js

```diff
@@ -175,53 +187,133 @@ PostmanCLIReporter = function (emitter, reporterOptions, options) {
             return;
         }
 
-        var timingTable,
-            timings,
-            timingPhases,
-            size = o.response && o.response.size(),
-            timingHeaders = {
-                prepare: 'prepare',
-                wait: 'wait',
-                dns: 'dns-lookup',
-                tcp: 'tcp-handshake',
-                secureHandshake: 'ssl-handshake',
-                firstByte: 'transfer-start',
-                download: 'download',
-                process: 'process',
-                total: 'total'
-            };
-
-        size = size && (size.header || 0) + (size.body || 0) || 0;
-
-        print.lf(colors.gray('[%d %s, %s, %s]'), o.response.code, o.response.reason(),
-            util.filesize(size), util.prettyms(o.response.responseTime));
-
-        // if there are redirects, get timings for the last request sent
-        timings = _.last(_.get(o, 'history.execution.data'));
-        timings = timings && timings.timings;
+        if (!(o.request && o.response)) {
+            print.lf(colors.red('[errored]'));
+            print.lf(colors.red('     %s'), 'Internal error! Could not read response data.');
 
-        // print timing info of the request
-        if (options.verbose && timings) {
-            timingPhases = util.beautifyTime(sdk.Response.timingPhases(timings));
+            return;
+        }
+
+        // quickly print out basic non verbose response meta and exit
+        if (!options.verbose) {
+            print.lf(colors.gray('[%d %s, %s, %s]'), o.response.code, o.response.reason(),
+                util.filesize(o.response.size().total), util.prettyms(o.response.responseTime));
 
-            timingTable = new Table({
-                chars: _.defaults({ mid: '', middle: '' }, cliUtils.cliTableTemplate_Blank),
-                colAligns: _.fill(Array(_.size(timingPhases)), 'left'),
-                style: { 'padding-left': 2 }
-            });
+            return;
+        }
 
-            timingPhases = _.transform(timingHeaders, (result, header, key) => {
+        // this point onwards the output is verbose. a tonne of variables are created here for
+        // keeping the output clean and readable
+
+        let req = o.request,
+            res = o.response,
+
+            // set values here with abundance of caution to avoid erroring out
+            reqSize = util.filesize(req.size().total),
+            resSize = util.filesize(res.size().total),
+            code = res.code,
+            reason = res.reason(),
+            mime = res.contentInfo() || {},
+            timings = _.last(_.get(o, 'history.execution.data')),
+
+            reqHeadersLen = _.get(req, 'headers.members.length'),
+            resHeadersLen = _.get(res, 'headers.members.length'),
+
+            resTime = util.prettyms(res.responseTime || 0),
+
+            reqText = (options.verbose && req.body) ? req.body.toString() : E,
```

Comment:
Note: `toString` don't work for `formdata`, `file`, and `graphql` modes.

I see we are not logging empty payloads. so no issues.